### PR TITLE
Vault-Controller-Strategy for Aave

### PR DESCRIPTION
**Pull request by Gwydce**

### CODE REVIEW

**Review by batudo — COMMENTED** [review]
I thought this was sent, but no, it got stuck on a draft... my bad.

I know this is probably not the thing your are working on now, but well... there you go =P

File: hardhat.config.js [hunk 1]
```diff
@@ -16,6 +16,51 @@ task('accounts', 'Prints the list of accounts', async () => {
...
  */
+
+const fs = require('fs');
+const accounts = JSON.parse(fs.readFileSync(".accounts"));
+const infuraKey = process.env.INFURA_KEY;
+
 module.exports = {
-  solidity: '0.8.4'
+  solidity: {
+    version: '0.8.3',
```
[the remark this answers is not in the document]
Comment (Gwydce) [reply hardhat.config.js:30]: Deployed with the 0.8.3, the 0.8.4 gave me a few "weird" compile errors.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# 2piFinance - Yield maximizer
+
+This repo contains the contracts deployed for our vaults and strategies.
+
+## The architecture
+Consist in 3 parts: vault <=> controller <=> strategy
+
+- Vault that receive the funds from users, and then send them to the controller.
+- Controller manage what is the currenct active strategy for a given token and it's the link between vault and the strategy
```
Comment (batudo) [inline README.md:9]:
```suggestion
- Controller manage what is the current active strategy for a given token and it's the link between vault and the strategy
```

File: hardhat.config.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 require('@nomiclabs/hardhat-waffle')
+require("@tenderly/hardhat-tenderly");
```
Comment (batudo) [inline hardhat.config.js:2]:
```suggestion
require('@tenderly/hardhat-tenderly')
```

File: hardhat.config.js [hunk 4]
```diff
@@ -10,12 +11,43 @@ task('accounts', 'Prints the list of accounts', async () => {
   }
 })
 
-// You need to export an object to set up your config
-// Go to https://hardhat.org/config/ to learn more
+const fs = require('fs');
```
Comment (batudo) [inline hardhat.config.js:14]:
```suggestion
const fs = require('fs')
```

File: hardhat.config.js [hunk 5]
```diff
@@ -10,12 +11,43 @@ task('accounts', 'Prints the list of accounts', async () => {
...
+const accounts = JSON.parse(fs.readFileSync(".accounts"));
```
Comment (batudo) [inline hardhat.config.js:15]:
```suggestion
const accounts = JSON.parse(fs.readFileSync('.accounts'))
```

File: hardhat.config.js [hunk 6]
```diff
@@ -10,12 +11,43 @@ task('accounts', 'Prints the list of accounts', async () => {
...
+    settings: {
+      optimizer: {
+        enabled: true,
+        runs: 10000
+      }
+    }
+  },
+  networks: {
+    polygon: {
+      url: "https://rpc-mainnet.maticvigil.com/",
```
Comment (batudo) [inline hardhat.config.js:33]:
```suggestion
      url: 'https://rpc-mainnet.maticvigil.com/',
```

File: hardhat.config.js [hunk 7]
```diff
@@ -10,12 +11,43 @@ task('accounts', 'Prints the list of accounts', async () => {
...
+      network_id: 137,
+      gas: 5e6, // gas estimate fails sometimes
+      confirmations: 2,
+      timeoutBlocks: 200,
+      skipDryRun: true
+    },
+    mumbai: {
+      url: "https://rpc-mumbai.maticvigil.com/",
+      // url: "https://matic-mumbai.chainstacklabs.com",
+      // url: "https://matic-testnet-archive-rpc.bwarelabs.com",
```
Comment (batudo) [inline hardhat.config.js:44]:
```suggestion
      url: 'https://rpc-mumbai.maticvigil.com/',
      // url: 'https://matic-mumbai.chainstacklabs.com',
      // url: 'https://matic-testnet-archive-rpc.bwarelabs.com',
```

**Review by batudo — APPROVED** [review]
👍

**Pull request merged**